### PR TITLE
Update TFM to netstandard2.0

### PR DESCRIPTION
**Pull request by jbagga**
cc @pranavkm @pakrym

### CODE REVIEW

File: build/dependencies.props [hunk 1]
```diff
@@ -5,7 +5,7 @@
     <AppInsightsVersion>2.1.0-beta3</AppInsightsVersion>
```
Comment (pranavkm) [inline build/dependencies.props]: Aren't you missing the change to `common.props`?

File: src/Microsoft.Web.Xdt.Extensions/Microsoft.Web.Xdt.Extensions.csproj [hunk 2]
```diff
@@ -4,7 +4,7 @@
 
   <PropertyGroup>
     <Description>Additional functionality for Xdt transforms.</Description>
-    <TargetFramework>net451</TargetFramework>
+    <TargetFramework>net461</TargetFramework>
```
Comment (pranavkm) [inline src/Microsoft.Web.Xdt.Extensions/Microsoft.Web.Xdt.Extensions.csproj]: List these guys here https://github.com/aspnet/Coherence-Signed/issues/526.

File: build/dependencies.props (hunk 1 above)
Comment (jbagga) [inline build/dependencies.props]: `common.props` doesn't have the $(TargetFrameworkIdentifier) like [this](https://github.com/aspnet/CORS/pull/112/files#diff-9780bf7a72094f1ac4b023e438b548baR19). Is there some other change as well?

Comment (JunTaoLuo) [inline build/dependencies.props]: I think the test project wouldn't be able to compile without the common.props change to add NS.Library.NETFramework.

File: test/Microsoft.AspNetCore.ApplicationInsights.HostingStartup.Tests/Microsoft.AspNetCore.ApplicationInsights.HostingStartup.Tests.csproj [hunk 3]
```diff
@@ -3,8 +3,9 @@
   <Import Project="..\..\build\common.props" />
 
   <PropertyGroup>
-    <TargetFramework>netcoreapp2.0</TargetFramework>
-    <PackageTargetFallback>$(PackageTargetFallback);portable-net40+sl5+win8+wp8+wpa81;portable-net45+win8+wp8+wpa81</PackageTargetFallback>
+    <TargetFrameworks>netcoreapp2.0;net461</TargetFrameworks>
+    <TargetFrameworks Condition=" '$(OS)' != 'Windows_NT' ">netcoreapp2.0</TargetFrameworks>
+    <PackageTargetFallback Condition=" '$(TargetFramework)' == 'netcoreapp2.0' ">$(PackageTargetFallback);portable-net40+sl5+win8+wp8+wpa81;portable-net45+win8+wp8+wpa81</PackageTargetFallback>
```
Comment (JunTaoLuo) [inline test/Microsoft.AspNetCore.ApplicationInsights.HostingStartup.Tests/Microsoft.AspNetCore.ApplicationInsights.HostingStartup.Tests.csproj]: Is this still needed and is the should the condition check for TargetFrameworks instead TargetFramework?

File: test/Microsoft.AspNetCore.AzureAppServicesIntegration.Tests/Microsoft.AspNetCore.AzureAppServicesIntegration.Tests.csproj [hunk 4]
```diff
@@ -3,8 +3,9 @@
   <Import Project="..\..\build\common.props" />
 
   <PropertyGroup>
-    <TargetFramework>netcoreapp2.0</TargetFramework>
-    <PackageTargetFallback>$(PackageTargetFallback);portable-net40+sl5+win8+wp8+wpa81;portable-net45+win8+wp8+wpa81</PackageTargetFallback>
+    <TargetFrameworks>netcoreapp2.0;net461</TargetFrameworks>
+    <TargetFrameworks Condition=" '$(OS)' != 'Windows_NT' ">netcoreapp2.0</TargetFrameworks>
+    <PackageTargetFallback Condition=" '$(TargetFramework)' == 'netcoreapp2.0' ">$(PackageTargetFallback);portable-net40+sl5+win8+wp8+wpa81;portable-net45+win8+wp8+wpa81</PackageTargetFallback>
```
Comment (JunTaoLuo) [inline test/Microsoft.AspNetCore.AzureAppServicesIntegration.Tests/Microsoft.AspNetCore.AzureAppServicesIntegration.Tests.csproj]: ditto

File: src/Microsoft.Web.Xdt.Extensions/Microsoft.Web.Xdt.Extensions.csproj (hunk 2 above)
Comment (Tratcher) [inline src/Microsoft.Web.Xdt.Extensions/Microsoft.Web.Xdt.Extensions.csproj]: Revert, this package has very specific target requirements.

File: src/Microsoft.AspNetCore.AzureAppServices.SiteExtension/Microsoft.AspNetCore.AzureAppServices.SiteExtension.csproj [hunk 5]
```diff
@@ -5,7 +5,7 @@
   <PropertyGroup>
     <title>ASP.NET Core Extensions</title>
     <Description>This extension enables additional functionality for ASP.NET Core on Azure WebSites, such as enabling Azure logging.</Description>
-    <TargetFramework>net451</TargetFramework>
+    <TargetFramework>net461</TargetFramework>
```
Comment (Tratcher) [inline src/Microsoft.AspNetCore.AzureAppServices.SiteExtension/Microsoft.AspNetCore.AzureAppServices.SiteExtension.csproj]: Revert.

File: test/Microsoft.AspNetCore.ApplicationInsights.HostingStartup.Tests/Microsoft.AspNetCore.ApplicationInsights.HostingStartup.Tests.csproj [hunk 6]
```diff
@@ -3,8 +3,9 @@
   <Import Project="..\..\build\common.props" />
 
   <PropertyGroup>
-    <TargetFramework>netcoreapp2.0</TargetFramework>
-    <PackageTargetFallback>$(PackageTargetFallback);portable-net40+sl5+win8+wp8+wpa81;portable-net45+win8+wp8+wpa81</PackageTargetFallback>
+    <TargetFrameworks>netcoreapp2.0;net461</TargetFrameworks>
+    <TargetFramework Condition=" '$(OS)' != 'Windows_NT' ">netcoreapp2.0</TargetFramework>
```
Comment (JunTaoLuo) [inline test/Microsoft.AspNetCore.ApplicationInsights.HostingStartup.Tests/Microsoft.AspNetCore.ApplicationInsights.HostingStartup.Tests.csproj]: Wait a second this looks wrong. You are setting the `TargetFrameworks` property but you are not overriding it when it's xplat, you are setting a different property `TargetFramework`.

File: test/Microsoft.AspNetCore.AzureAppServicesIntegration.Tests/Microsoft.AspNetCore.AzureAppServicesIntegration.Tests.csproj [hunk 7]
```diff
@@ -3,8 +3,9 @@
   <Import Project="..\..\build\common.props" />
 
   <PropertyGroup>
-    <TargetFramework>netcoreapp2.0</TargetFramework>
-    <PackageTargetFallback>$(PackageTargetFallback);portable-net40+sl5+win8+wp8+wpa81;portable-net45+win8+wp8+wpa81</PackageTargetFallback>
+    <TargetFrameworks>netcoreapp2.0;net461</TargetFrameworks>
+    <TargetFramework Condition=" '$(OS)' != 'Windows_NT' ">netcoreapp2.0</TargetFramework>
```
Comment (JunTaoLuo) [inline test/Microsoft.AspNetCore.AzureAppServicesIntegration.Tests/Microsoft.AspNetCore.AzureAppServicesIntegration.Tests.csproj]: ditto

File: test/Microsoft.AspNetCore.ApplicationInsights.HostingStartup.Tests/Microsoft.AspNetCore.ApplicationInsights.HostingStartup.Tests.csproj (hunk 3 above)
Comment (jbagga) [inline test/Microsoft.AspNetCore.ApplicationInsights.HostingStartup.Tests/Microsoft.AspNetCore.ApplicationInsights.HostingStartup.Tests.csproj]: Okay, I can test this out once Mvc migration is checked in and make the necessary changes.

**Comment by JunTaoLuo** [comment]
I think the mvc updates have been merged, this is unblocked.

**Comment by jbagga** [comment]
@JunTaoLuo @pranavkm Updated

**Pull request merged**